### PR TITLE
imxrt/pit: Modify lifetime timer support

### PR DESCRIPTION
**Pull request by yashwanth686007**
This patch modifies lifetime timer support for imxrt.

Signed-off-by: Yashwanth <v.yashwanth@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d0e342cfb44a01b1e00773c2c81e26e8d4b70cba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d0e342cfb44a01b1e00773c2c81e26e8d4b70cba] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_pit.c [hunk 1]
```diff
@@ -554,24 +556,29 @@ static int imxrt_pit_ioctl(struct timer_lowerhalf_s *lower, int cmd,
...
 	case TCIOC_GETLIFETIME:
 		freq = imxrt_clock_getperclkfreq();
 		ticks = imxrt_pit_getlifetimetimercount(priv->pit->base);
-		ret = ((uint32_t)USEC_PER_SEC * (uint32_t)ticks) / freq;
-		tmrvdbg("lifetime counter %d (ms)\n", ret);
+		total_time = ((ULLONG_MAX - ticks) * USEC_PER_SEC) / freq;
+
+		total_time = total_time / USEC_PER_SEC;
+
+		l_time = localtime((const time_t *)&total_time);
```
Comment (NitishAmbastha) [inline os/arch/arm/src/imxrt/imxrt_pit.c]: time_t internally uses uint32_t in TizenRT. If we use libc functions here, possible loss of data?

Comment (yashwanth686007) [reply os/arch/arm/src/imxrt/imxrt_pit.c]: The value with 32 bit counter can go beyond 100 years. I think it should be fine.

Comment (NitishAmbastha) [reply os/arch/arm/src/imxrt/imxrt_pit.c]: agree

**Pull request merged**